app/controllers/api/v1/packages_controller.rb

```diff
@@ -36,8 +36,14 @@ class PackagesController < Api::V1::ApiController
 
       api :GET, "/v1/packages", "get all packages for the item"
       def index
-        @packages = @packages.find(params[:ids].split(",")) if params[:ids].present?
-        render json: @packages, each_serializer: serializer, include_orders_packages: true
+        if params["browseSearch"]
```

Comment:
I don't think this property is needed, you can just check the presence of `params["searchText"`] instead.

If we look at the api design we **want**,  I personally would expect all the following to work :

- `/packages` -> all
- `/packages?searchText=text` -> search
- `/packages?ids=1,2,3` -> specific ids

for which the implementation could *roughly* be something like : 
```
records = @packages
records = records.search({search_text: params['searchText']}) if params["searchText"].present?
records = records.find(params[:ids].split(",")) if params[:ids].present?
records = records..filter('state' => params["state"]) if params["state"].present?
```

One question for @steveyken --> I kinda feel we should consolidate the serializers, and not have `BrowsePackageSerializer`. All of those `is_browse_app` make the control-flow complexity sky-rocket.
I've already started using the normal serializer on browse (for some requests), they're pretty much the same

Reply:
yes that's good. re ```BrowsePackageSerializer``` I think @abhinav had begun work a while back to see if it was possible. Did you find any roadblocks?

Reply:
Yes @steveyken I tried doing that once for message serializers, but it wasn't required at that time, so I stopped working on it.